src/schema/schema_open.c

```diff
@@ -297,8 +297,19 @@ __schema_open_index(WT_SESSION_IMPL *session,
 	for (i = 0; ret == 0; i++, ret = cursor->next(cursor)) {
 		WT_ERR(cursor->get_key(cursor, &uri));
 		name = uri;
-		if (!WT_PREFIX_SKIP(name, tmp->data))
+
+		if (!WT_PREFIX_SKIP(name, tmp->data)) {
+			/*
+			 * We reached the end of index list, remove the rest of
+			 * in memory indices, they no longer exist.
+			 */
+			while (i < table->nindices) {
+				WT_ERR(__wt_schema_destroy_index(session,
```

Comment:
Don't we want to proceed even if `__wt_schema_destroy_index()` returns an error, that is, continue freeing the memory? Otherwise I think we'll leak it. (That's what we do in `__wt_schema_close_table()`.)

I don't feel strongly.